mlrun/runtimes/base.py

```diff
@@ -510,7 +510,12 @@ def run(
             # single run
             try:
                 resp = self._run(run, execution)
-                if watch and mlrun.runtimes.RuntimeKinds.is_watchable(self.kind):
+                if (
+                    watch
+                    and mlrun.runtimes.RuntimeKinds.is_watchable(self.kind)
+                    # API shouldn't watch logs, its the client job to query the run logs
+                    and not mlrun.config.is_running_as_api()
```

Comment:
Not much to add to Saar's comments. You could set watch to false at the top of the function if in API instead of adding it to the if which IMO will be a bit cleaner and might catch if someone else is using watch elsewhere.